README.md

```diff
@@ -1,15 +1,146 @@
+Bazel-integration-testing
+=================================
+**Because your build infra also needs build (and tests)**
+
 Bazel CI | Travis CI
 :---: | :---:
 [![Build status](https://badge.buildkite.com/b0041826d71f5484c22145f44b3eac12357f51feb6ba6abb57.svg?branch=master)](https://buildkite.com/bazel/bazel-integration-testing-postsubmit) | [![Build Status](https://travis-ci.org/bazelbuild/bazel-integration-testing.svg?branch=master)](https://travis-ci.org/bazelbuild/bazel-integration-testing)
 
-# Integration test framework for end to end test of Bazel utilities
 
+## Problem statement  
+Bazel-integration-testing aims to give confidence when developing code that is strongly dependent on Bazel.  
+It does so by allowing you to have **isolated reproducible tests which run Bazel inside of them**.  
+Happy path testing are often possible inside of bazel by adding examples of your usage but failing cases are much harder.      
+Few examples for such code can be Starlark plugins (both open and closed source), tools that augment Bazel/Bazel query and actually also Bazel itself.    
+This project was inspired by Bazel's own integration tests which were in bash at the time and were coupled inside.    
+Internally at Wix we use this to develop and test several internal rules, macros and also tools that need to run `bazel query` and `bazel`.  
+ADD OSS EXAMPLE  
+## Target audience  
+Bazel Starlark rule developers (both open source and closed source)  
+Bazel ecosystem tool developers (both open source and closed source)   
+Bazel developers  
+## Alternatives
+For integration testing as of Jan'19 there is no alternative which is usable outside of Bazel.  
+For some use-cases and tests one can use bazel-skylib's [unittest](https://github.com/bazelbuild/bazel-skylib/blob/master/lib/unittest.bzl) framework.  
+This is faster and might be easier to setup in the test (smaller scope) but has the built-in limitations of unit testing where you make assumptions about the environment.  
+To be clear it's usually a good idea to have a mix of unit tests and integration tests for your feedback loop and confidence.
+## Architecture  
+The project is built from repository rules, build rules and test drivers.  
+The build rules are per stack (currently JVM, go, python) and are essentially a wrapper over the `$stack_test` rule. For every bazel version configured such a wrapper is generated along with a `test_suite` aggregate.  
+The JVM build rules are a bit more advanced in that they integrate with `bazel_external_dependency_archive` repository rule to support external dependencies in the test workspace.  
+Lastly each stack has a specific test driver which eases creation of scratch WORKSAPCE files, BUILD files, source files, running of Bazel and more.                                                                                                                                                                                                 
+## Getting started
```

Comment:
Since the getting started is per language what would you think about breaking this up into a list of links to a go/python stub in the go and python dirs, respectively, and then these contents in the `java/build/bazel/...` dir?

Reply:
moved